### DISCUSSION TITLE
Invalid token: token is expired

### DISCUSSION THREAD

**Discussion by sroussey** [supabase/supabase#1884] [Questions]
Looking at the gotrue-js code, it looks like it will rotate tokens when expired, but I don't see that happening. Not sure what to do. I am getting this error on an API endpoint that where I call supabase.auth.api.getUser. When I look at the gotrue code it looks like it should have set a timer to get a new token, but clearly that is not working.

**Comment by sroussey** [comment]
So every time my computer goes to sleep or don't visit the site for a while I get this error. Actually, same experience on app.supabase.io where I hit reload in both cases to fix the issue. I don't want my user to have to reload every time they are away for a bit.

**Comment by silentworks** [comment]
This is something you will probably need to handle application side. You could open a modal telling the user their session has expired or something and they should click this button to stay logged in. JWTs require an expiry time so this is unavoidable.

**Reply by sroussey** [reply]
But that is why there is an option in the gotrue code to renew the token in the background. Why is it not working?
